### PR TITLE
Implement final mapping for plugin

### PR DESCRIPTION
**Pull request by lzap**
Implements the final mapping after discussion at:

https://community.theforeman.org/t/new-config-report-summary-columns/26531

See the original post for the change, nochange and failure mapping as well as keywords.

After merge, make sure to cleanup your DB:

```
HostReport.delete_all
ReportKeywords.delete_all
```

Then re-upload tests with smart proxy and its patch: https://github.com/theforeman/smart_proxy_reports/pull/20

@ofedoren @ron-lavi

### CODE REVIEW

File: app/controllers/api/v2/host_reports_controller.rb [hunk 1]
```diff
@@ -34,10 +34,9 @@ def show
           param :host, String, required: true, desc: N_("Hostname of the report's host origin")
           param :format, HostReport.formats.keys, required: false, desc: N_('Format of the report, e.g. Ansible')
           param :reported_at, String, required: true, desc: N_('UTC time of the report')
-          param :applied, Integer, required: false, desc: N_('Number of applied resources or tasks')
-          param :failed, Integer, required: false, desc: N_('Number of failed resources or tasks')
-          param :pending, Integer, required: false, desc: N_('Number of pending resources or tasks')
-          param :other, Integer, required: false, desc: N_('Number of other resources or tasks')
+          param :change, Integer, required: false, desc: N_('Summary count of actions with change (semantic is different for each report type)')
```
Comment (ofedoren) [inline app/controllers/api/v2/host_reports_controller.rb]:
```suggestion
          param :change, Integer, required: false, desc: N_('Summary count of actions with change (semantics is different for each report type)')
```

File: test/controllers/api/v2/host_reports_controller_test.rb [hunk 2]
```diff
@@ -119,114 +118,80 @@ def test_create_invalid
...
-          applied: 5, failed: 1, pending: 1, other: 0
+          change: 1, nochange: 2, failure: 3
         },
       }
       assert_response :forbidden
     end
 
-    test 'when "require_ssl_smart_proxies" is true and "require_ssl" is false, HTTP requests should be able to create reports' do
-      # since require_ssl_smart_proxies is only applicable to HTTPS connections, both should be set
+    test 'when "require_ssl" is true , HTTP requests should be able to create reports' do
```
Comment (ofedoren) [inline test/controllers/api/v2/host_reports_controller_test.rb]:
```suggestion
    test 'when "require_ssl" is false , HTTP requests should be able to create reports' do
```

**Review by ofedoren — COMMENTED** [review]
Thanks, @lzap! LGTM, just few nitpicks.

Also, if you updating JS files, could you please run `npm run test -- -u` to update the snapshots?

**Comment by lzap** [comment]
All done, pushed.

**Comment by lzap** [comment]
I don't understand the errors, this gem is not supposed to do any kind of YAML loading. Yet:

```
Error:
Api::V2::HostReportsControllerTest::Non Admin User#test_0001_store statuses:
Psych::DisallowedClass: Tried to load unspecified class: Time
    test/test_helper.rb:68:in `before_setup'
```

**Comment by lzap** [comment]
Looks like psych 4.0 vs rails 6.0 is causing this: https://github.com/theforeman/foreman/pull/9012/files removed rdoc dependency (might need to pin it not sure).

**Comment by lzap** [comment]
Looks like I have to pin rdoc too even I am not pulling it. Not sure, hopefully this works now.

File: foreman_host_reports.gemspec [hunk 3]
```diff
@@ -16,5 +16,7 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = '>= 2.5.0'
 
-  s.add_development_dependency 'rdoc'
+  # Pin rdoc, which pulls in psych 4.0 since 6.4.0, but
+  # Rails 6.0 does not work with psych 4.0 (https://github.com/theforeman/foreman/pull/9012)
+  s.add_development_dependency 'rdoc', '< 6.4.0'
```
Comment (evgeni) [inline foreman_host_reports.gemspec]: with the changes from #16, this isn't needed anymore.

(it doesn't hurt tho)

Comment (lzap) [reply foreman_host_reports.gemspec]: Oh really? So delete completely? Nice.

**Comment by lzap** [comment]
Rebased, trying ONCE AGAIN and crossing my fingers.

**Comment by lzap** [comment]
Rebased on top of Ansible UI changes.

**Review by ofedoren — CHANGES_REQUESTED** [review]
Thanks, @lzap! LGTM, only one small change for now just to make it work again:

Please change https://github.com/theforeman/foreman_host_reports/blob/b37a53b519fb8547d5f3a48dd2b2216eeeb31204/webpack/src/Router/HostReports/IndexPage/Components/HostReportsTable/Components/StatusCell.js#L13 to be 'failure'

**Comment by lzap** [comment]
Thanks! Pushed.

**Review by ofedoren — APPROVED** [review]
Thanks, ACK, waiting for https://github.com/theforeman/smart_proxy_reports/pull/20 to be merged first.

**Pull request merged**